File: problems/005/p005.c

#include <stdio.h>

int resolve(int factor)
{

	int smallest_multiple = 0;
	int num = factor * factor;
	while (smallest_multiple == 0)
	{

		int match = 1;
		for (int i = 2; i <= factor; i++)
		{

			if (num % i !=0)
			{
				match = 0;
				break;
			}
		}

		if (match)
		{
			smallest_multiple = num;
		}

		num++;
	}

	return smallest_multiple;
}

int main()
{

	int result = resolve(20);

	printf("Result=%d\n", result);

	return 0;
}

File: problems/005/p005.go
package main

import "fmt"

func main() {

	var result int =  resolve(20)

	fmt.Printf("Result=%d\n", result)

}

func resolve(factor int) int {

	var smallest_multiple int
	num := factor
	for smallest_multiple == 0 {

		var match = true
		for i := 2; i <= factor; i++ {

			if num % i != 0 {
				match = false;
				break;
			}

		}

		if match {
			smallest_multiple = num
		}

		num++
	}

	return smallest_multiple
}
